packages/ec-core/src/operator/genome_extractor.rs

```diff
@@ -1,20 +1,67 @@
+use std::convert::Infallible;
+
 use super::{Composable, Operator};
 use crate::individual::Individual;
 
-pub struct GenomeExtractor;
-
-/// Returns a _cloned_ copy of the given individual's genome.
+/// An [`Operator`] that returns a _cloned_ copy of the given [`Individual`]'s
+/// genome.
 ///
 /// This copy of the genome is cloned so that subsequent stages of a pipeline
 /// (like mutation) can take ownership and mutate as appropriate without
 /// affecting the parent individual.
+///
+/// # Examples
+///
+/// In this example we chain `GenomeExtractor` with a simple mutator to
+/// extract the genome from an individual and mutate it.
+///
+/// ```
+/// # use rand::{rngs::ThreadRng, thread_rng, Rng};
+/// # use ec_core::{
+/// #     individual::ec::EcIndividual,
+/// #     operator::{
+/// #         genome_extractor::GenomeExtractor,
+/// #         mutator::{Mutate, Mutator},
+/// #         Composable, Operator,
+/// #     },
+/// # };
+/// // Flip a single bit in an array of booleans.
+/// struct FlipOne;
+///
+/// impl Mutator<[bool; 4]> for FlipOne {
+///     fn mutate(&self, mut genome: [bool; 4], rng: &mut ThreadRng) -> anyhow::Result<[bool; 4]> {
+///         let index = rng.gen_range(0..genome.len());
+///         genome[index] = !genome[index];
+///         Ok(genome)
+///     }
+/// }
+///
+/// let genome = [true, false, false, true];
+/// let individual = EcIndividual::new(genome, ());
+///
+/// let mutate = Mutate::new(FlipOne);
+/// let chain = GenomeExtractor.then(mutate);
+///
+/// let result = chain.apply(&individual, &mut thread_rng()).unwrap();
```

Comment:
Do we rather want to use `?` error handling in doctests? I think it's more ideomatic and less clutter for the example.

Reply:
Well we could explicitly write the entire function and hide that, but just putting a # Ok::<_,T>(()) at the end is probably the simplest solution